src/entities/trades/swapr-v3/SwaprV3.ts

```diff
@@ -105,7 +119,52 @@ export class SwaprV3Trade extends TradeWithSwapTransaction {
           quoteCurrency.name,
         )
 
-    const routes = await getRoutes(setToken, quoteToken, chainId)
+    const routes = isTradeExactInput
+      ? await getRoutes(setToken as UniswapToken, quoteToken as UniswapToken, chainId)
+      : await getRoutes(quoteToken as UniswapToken, setToken as UniswapToken, chainId)
+
+    const quoteParams = routes.map((route) => [
+      encodeRouteToPath(route, !isTradeExactInput),
+      `0x${amount.raw.toString(16)}`,
+    ])
+    const DEFAULT_GAS_QUOTE = 2_000_000
```

Comment:
why this value as default gas quote?

Reply:
I think this is necessary to avoid "running out of gas" error. This way we say we have enough gas to run the simulation, which will return an answer we use later.

Reply:
I think it's a good enough amount, but not sure why this number. This comes from algebra logic.